src/library_browser.js

```diff
@@ -1164,6 +1164,12 @@ var LibraryBrowser = {
       GL.newRenderingFrameStarted();
 #endif
 
+      // If the current GL context is an OffscreenCanvas, but it was initialized with implicit swap mode, perform the swap
+      // in behalf of the user.
+      if (typeof GL !== 'undefined' && GL.currentContext && !GL.currentContext.attributes.explicitSwapControl && GL.currentContext.GLctx.commit) {
```

Comment:
Yeah, that is intended. If the context is based on an OffscreenCanvas, the `commit` function exists, and when it exists, the context always has explicit swap control. If the user wanted to have implicit swap control behavior, then we do the swap here. This code path could be hit either in main thread or a pthread.

In main thread it can occur in two different ways:
   - if the canvas has been transferred to a pthread at least once during the application lifetime and then the pthread has quit and we are again rendering in the main thread, with desired implicit swap control, or
   - if in the main thread user has created a context with explicit swap control, deleted it, and then later created a new context with implicit swap control

This is because in the current form of the spec, if the control of a canvas's rendering has been once transferred offscreen, it can't be transferred back. This means that after transferring, we always need to use OffscreenCanvas to render, i.e. always will have explicit `.commit`.